homeassistant/components/adax/climate.py

```diff
@@ -164,3 +180,12 @@ async def async_update(self) -> None:
         self._attr_target_temperature = data["target_temperature"]
         self._attr_current_temperature = data["current_temperature"]
         self._attr_available = self._attr_current_temperature is not None
+        if data["target_temperature"] == 0:
+            self._attr_hvac_mode = HVACMode.OFF
+            self._attr_icon = "mdi:radiator-off"
+            if self._attr_target_temperature == 0:
+                self._attr_target_temperature = self.min_temp
```

Comment:
```suggestion
            self._attr_target_temperature = self.min_temp
```

Reply:
There is a bug in the code that I discovered through this suggestion.

 self._attr_target_temperature is keeping track of the temperature when the heater is off (i.e. when the heater is reporting 0), so that when it is turned on it is turned on at the last set target temperature. Will make a new commit that fixes this.